rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -29,13 +32,68 @@ local function fetch_request_body()
   return body
 end
 
+function _M.get_pem_cert_key(hostname)
+  return certificate_data:get(hostname)
+end
+
+local function handle_servers()
+  if ngx.var.request_method ~= "POST" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only POST requests are allowed!")
+    return
+  end
+
+  local raw_servers = fetch_request_body()
+
+  local ok, servers = pcall(json.decode, raw_servers)
+  if not ok then
+    ngx.log(ngx.ERR,  "could not parse servers: " .. tostring(servers))
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
+
+  local err_buf = {}
+  -- Update certificates and private keys for each host
```

Comment:
I don't think this comment adds any value. If you think it's hard to follow the logic below then I'd suggest you extract it into a separate function with descriptive name.